components/channel_header/channel_header.js

```diff
@@ -598,6 +599,11 @@ class ChannelHeader extends React.PureComponent {
         if (rhsState === RHSStates.FLAG) {
             flaggedIconClass += ' active';
         }
+        const count = this.props.pinnedPostsCounts;
+        let countText = '-';
+        if (count > 0) {
+            countText = count.toString();
+        }
```

Comment:
this might be simpler, logic is good but it might be easier to read and also takes care of anything `falsy`
```js
const countText = this.props.pinnedPostsCount ? this.props.pinnedPostsCount.toString() : '-';
```